### PR TITLE
Bring your own certs docs

### PR DESCRIPTION
**Pull request by jtudelag**
Addresses https://github.com/bitnami-labs/sealed-secrets/issues/337

### CODE REVIEW

File: docs/bring-your-own-certificates.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# Bring your own certificates
+
+The controller generates its own certificates when is deployed for the first time, it also manages the renewal for you.
+But you can also bring your own certificates so the controller can consume them as well.
+
+The controller consumes certificates contained in any secret labeled with `sealedsecrets.bitnami.com/sealed-secrets-key=active`,
+the secret has to live in the same namespace as the controller. There can be multiple such keys.
```
Comment (mkmik) [inline docs/bring-your-own-certificates.md:7]:
```suggestion
the secret has to live in the same namespace as the controller. There can be multiple such secrets.
```

File: docs/bring-your-own-certificates.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+## Create a tls k8s secret, using your recently created RSA key pair:
+```bash
+kubectl -n "$NAMESPACE" create secret tls "$SECRETNAME" --cert="$PUBLICKEY" --key="$PRIVATEKEY"
+kubectl -n "$NAMESPACE" label secret "$SECRETNAME" sealedsecrets.bitnami.com/sealed-secrets-key=active
+```
+
+## Deleting the controller Pod is needed to pick they new keys:
+```bash
+kubectl -n  "$NAMESPACE" delete pod -l app.kubernetes.io/name=sealed-secrets
```
Comment (mkmik) [inline docs/bring-your-own-certificates.md:32]:
```suggestion
kubectl -n  "$NAMESPACE" delete pod -l name=sealed-secrets-controller
```

(At some point we tried the app.kubernetes.io/name label but we had to roll that back due to upgrade regressions.
I'll try again to migrate to the standard labels once I find a way to ensure we don't break the upgrade path.)

File: docs/bring-your-own-certificates.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+```
+
+## See the new certificates (private keys) in the controller logs:
+```bash
+kubectl -n "$NAMESPACE" logs $(kubectl -n  "$NAMESPACE" get pod -l app.kubernetes.io/name=sealed-secrets -o name | cut -d '/' -f2)
```
Comment (mkmik) [inline docs/bring-your-own-certificates.md:37]:
```suggestion
kubectl -n "$NAMESPACE" logs -l name=sealed-secrets-controller
```

File: docs/bring-your-own-certificates.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+## Used your recently created public key to "seal" your secret:
+Use your own certificate (key) by using the `--cert` flag:
+```bash
+kubeseal --cert "./${PUBLICKEY}" --scope cluster-wide < mysecret.yaml | kubectl apply -f-
+```
+
+## We can see the secret has been unsealed succesfully:
+```bash
+kubectl -n "$NAMESPACE" logs $(kubectl -n "$NAMESPACE" get pod -l app.kubernetes.io/name=sealed-secrets -o name | cut -d '/' -f2)
```
Comment (mkmik) [inline docs/bring-your-own-certificates.md:58]:
```suggestion
kubectl -n "$NAMESPACE" logs -l name=sealed-secrets-controller
```

File: docs/bring-your-own-certificates.md (hunk 2 above)
Comment (jtudelag) [reply docs/bring-your-own-certificates.md:32]: Sorry, but I dont get you...by default these are the [default Pod labels](https://github.com/helm/charts/blob/master/stable/sealed-secrets/templates/deployment.yaml#L25) you get, with the latest chart:

```
oc get pods --show-labels | grep seale
sealed-secrets-644b4c58bf-wf7kp                  1/1     Running     0          3h30m   app.kubernetes.io/instance=RELEASE-NAME,app.kubernetes.io/name=sealed-secrets,pod-template-hash=644b4c58bf
```

Also, `name=` is not a default label, it should be a [custom Pod label](https://github.com/helm/charts/blob/master/stable/sealed-secrets/values.yaml#L61)...

Comment (jtudelag) [reply docs/bring-your-own-certificates.md:32]: Sorry, my bad, I get it now: https://github.com/bitnami-labs/sealed-secrets/releases/download/v0.12.4/controller.yaml

The helm chart is not updated I would say...

**Comment by jtudelag** [comment]
Fixed changes.

**Comment by mkmik** [comment]
thanks again!

**Pull request merged**